### DISCUSSION TITLE
RTCDatachannel BufferedAmount property

### DISCUSSION THREAD

**Discussion by LoisBN** [flutter-webrtc/flutter-webrtc#783] [Q&A] [answered]
#### Hello flutter-webrtc team,

I am currently implementing file transfer features in my app but it seems that **there is no bufferedAmount** property for the RTCDatachannel class.

Are there any plans to add this property in the next updates of this library ?

**Comment by cloudwebrtc** [comment, marked as answer]
If there is a `bufferedAmount` parameter in the native API, I think we can implement it in the future

**Comment by simonbengtsson** [comment]
Not entirely related but in terms of speed of file transfer I found message size to more important than bufferedAmount. I use the following code in the recently open sources AirDash app to get this value:

https://github.com/simonbengtsson/airdash/blob/main/lib/transfer/data_sender.dart#L73

```dart
  int parseMaximumSize(RTCSessionDescription? description) {
    var remoteLines = description?.sdp?.split('\r\n') ?? [];

    int remoteMaximumSize = 0;
    for (final line in remoteLines) {
      if (line.startsWith('a=max-message-size:')) {
        var string = line.substring('a=max-message-size:'.length);
        remoteMaximumSize = int.parse(string);
        break;
      }
    }

    // 16 kb should be supported on all clients so we can use it
    // even if no max message is set
    return max(remoteMaximumSize, 16000);
  }
```
